animation_nodes/data_structures/attributes/attribute.pyx

```diff
@@ -0,0 +1,86 @@
+from .. lists.base_lists cimport (
+    LongList,
+    FloatList,
+    ColorList,
+    BooleanList,
+    Vector2DList,
+    Vector3DList,
+)
+
+cdef class Attribute:
+    def __cinit__(self, str name = None,
+                        AttributeType type = CUSTOM,
+                        AttributeDomain domain = POINT,
+                        AttributeDataType dataType = FLOAT,
+                        data = None):
+
+        self.name = name
+        self.type = type
+        self.domain = domain
+        self.dataType = dataType
+        self.data = data
+
+    def copy(self):
+        return Attribute(self.name, self.type, self.domain, self.dataType,
+                         self.data.copy())
+
+    def extend(self, data):
+        self.data.extend(data)
+
+    def repeated(self, Py_ssize_t amount = -1):
```

Comment:
This should repeat the attribute in-place. It should be called `replicate`. And the amount should not have a default value, -1 doesn't make sense here.

Reply:
Okay.